src/polygon.cpp

```diff
@@ -296,16 +296,10 @@ bool SEdgeList::AssemblePolygon(SPolygon *dest, SEdge *errorAt, bool keepDir) co
 // but they are considered to cross if they are coincident and overlapping.
 // If pi is not NULL, then a crossing is returned in that.
 //-----------------------------------------------------------------------------
-int SEdgeList::AnyEdgeCrossings(Vector a, Vector b,
-                                Vector *ppi, SPointList *spl) const
-{
-    int cnt = 0;
-    for(const SEdge *se = l.First(); se; se = l.NextAfter(se)) {
-        if(se->EdgeCrosses(a, b, ppi, spl)) {
-            cnt++;
-        }
-    }
-    return cnt;
+int SEdgeList::AnyEdgeCrossings(Vector a, Vector b, Vector *ppi, SPointList *spl) const {
+    auto cnt = std::count_if(l.begin(), l.end(),
+                             [&](SEdge const &se) { return se.EdgeCrosses(a, b, ppi, spl); });
+    return static_cast<int>(cnt);
```

Comment:
Can we not use `static_cast` for trivial integer casts?

Reply:
as in, you'd prefer `(int)cnt` ?

Reply:
Yep. I think e.g. `const_cast` makes sense when you're making it explicit that you're removing a `const` modifier, but doing simple integer casts with `static_cast` is just noise.

Reply:
Although in this case maybe the function just needs to return `size_t`.

Reply:
yeah, there are a bunch of places where it would make sense to return size_t but we return int.

Reply:
I'm fine fixing them to return `size_t`.